packages/discord.js/src/structures/User.js

```diff
@@ -188,12 +198,35 @@ class User extends Base {
   }
 
   /**
-   * The Discord "tag" (e.g. `hydrabolt#0001`) for this user
+   * The tag of this user
+   * <info>`hydrabolt#0001`, or `@hydrabolt` if they have been migrated to the new username system</info>
    * @type {?string}
    * @readonly
    */
   get tag() {
-    return typeof this.username === 'string' ? `${this.username}#${this.discriminator}` : null;
+    return typeof this.username === 'string'
+      ? this.hasNewUsername
+        ? `@${this.username}`
+        : `${this.username}#${this.discriminator}`
+      : null;
+  }
+
+  /**
+   * Indicates whether the user has been migrated to the new username system
+   * @type {boolean}
+   * @readonly
+   */
+  get hasNewUsername() {
```

Comment:
Can't say I'm a fan of this boolean. It's unfortunately that we're going to have an awkward middle ground where we have to support the transition, but exposing this as a getter that we KNOW is going to be removed after a few weeks seems like a very short-sighted addition.

Reply:
Now it's `User#usingLegacyUsername`